Prevent container from being wider than screen, and its left margin from becoming negative on narrow screens

sakerattminnas opened the pull request:
Prevents popup from being wider than the viewport.

Fixes: #36141 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<img width="1126" height="768" alt="Skärmbild 2025-09-22 185059" src="https://github.com/user-attachments/assets/c15598c8-8725-412e-b573-74d948e0ad6f" />
<img width="2307" height="783" alt="Skärmbild 2025-09-22 185753" src="https://github.com/user-attachments/assets/d96dd595-276e-45da-8bad-4eb48bcee866" />

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
Welcome @sakerattminnas!

Can you post before/after screenshots as detailed in our contributing guide? https://github.com/zulip/zulip/pulls/karlstolley has some good examples to crib from of how to make it really easy to see.

And how did you decide whether to use `em` or `px` for the numbers here? One can change font size, e.g. in the top-right menu, so it's good to test the 12px and 20px ends of the range for changes like this.

sayamsamal commented on web/styles/zulip.css:
```diff
@@ -302,8 +302,9 @@ p.n-margin {
     top: 0;
-    left: calc(50vw - 220px);
+    left: max(0px, calc(50vw - 220px));
     padding: 15px;
```
Let's use `box-sizing: border-box;` to account for the padding in the content width.

sakerattminnas commented:
`box-sizing: border-box;` is a great fix, I've changed the code to use it now

sakerattminnas commented:
I have changed the following after the last review, according to feedback:

- The container now uses ´box-sizing: border-box;´ to account for the padding when centering it.
- A wrapper div has been added, to make it possible to use `transform: translateX(...);` to center, while still using `translateY` for the animation.
- The css that concerns the positioning of the element has been moved so that it only applies to the wrapper div.
- The width of the container surrounding the wrapper div has been changed to `100vw` to avoid having to center it, as that was the problem that the wrapper was supposed to solve. This container is not visible and it is still possible to click elements underneath it.

The thing that prevents the popup from being too wide is the added `max-width: 100wv;`.

The screenshots in the original PR are still representative of what it looks like.

I would appreciate another review!

sakerattminnas commented:
I have now done the following, after the last feedback:
- Changed the name of the wrapper div to feedback-container-content-wrapper
- Set the max width of the content wrapper to `calc(100% - 20px)` instead of `100vw` to have some gap next to the popup
- Changed the unit of the max width of the outer container to `%` instead of `vw` to avoid offsetting the container slightly to the right

I would appreciate another review!

New after-screenshots:

<img  height="450" alt="image" src="https://github.com/user-attachments/assets/4fd0abc6-fd79-4a96-9436-773e7c2dfe3f" />

<img width="1913" height="913" alt="image" src="https://github.com/user-attachments/assets/97566969-2ee9-446d-b21e-7f5e5d299436" />

sakerattminnas commented on web/styles/zulip.css:
```diff
@@ -298,16 +298,23 @@ p.n-margin {
+#feedback_container {
+    display: none;
+    position: fixed;
+    width: 100vw;
```
Done!

sakerattminnas commented on web/styles/zulip.css:
```diff
@@ -298,16 +298,23 @@ p.n-margin {
     width: 400px;
-    top: 0;
-    left: calc(50vw - 220px);
+    max-width: 100vw;
```
Done!

sakerattminnas commented on web/styles/zulip.css:
```diff
@@ -298,16 +298,23 @@ p.n-margin {
 
-#feedback_container {
-    display: none;
+#feedback-container-visible {
```
Done!
